### PR TITLE
fixes #8471 - adds a number field input support

### PR DESCRIPTION
**Pull request by ohadlevy**
- change subnet vlan to number field
- change ldap port to number field

### CODE REVIEW

File: app/views/auth_source_ldaps/_form.html.erb [hunk 1]
```diff
@@ -15,7 +15,7 @@
       <%= text_f f, :name %>
       <%= text_f f, :host %>
       <%= checkbox_f(f, :tls, {:label => "LDAPS", :onchange => 'change_ldap_port(this)'}) %>
-      <%= text_f f, :port, :data => {:default_ports => AuthSourceLdap::DEFAULT_PORTS} %>
+      <%= number_f f, :port, :min => 1, :data => {:default_ports => AuthSourceLdap::DEFAULT_PORTS } %>
```
Comment (dLobatog) [inline app/views/auth_source_ldaps/_form.html.erb]: `:max => 65535`?

File: app/helpers/layout_helper.rb [hunk 2]
```diff
@@ -310,4 +310,13 @@ def trunc(text, length = 32)
...
     button_tag(text, :class => 'btn btn-default', :data => { :dismiss => data_dismiss })
   end
+
+  def number_f(f, attr, options = {})
+    field(f, attr, options) do
+      addClass options, "form-control"
+      f.number_field attr, options
+    end
+  end
+
```
Comment (dLobatog) [inline app/helpers/layout_helper.rb]: There's an extra line (or 2) unneeded here

**Comment by dLobatog** [comment]
@ohadlevy Tested and it's almost ready, unless I'm wrong with the max port thing, thanks!

**Comment by ohadlevy** [comment]
not sure why its failing, another [test] to double check

**Comment by dLobatog** [comment]
@ohadlevy [test] once again, it only fails on mysql with `07:56:51 hudson.AbortException: No test report files were found. Configuration error?` before even running the actual tests

**Comment by domcleal** [comment]
Due to http://projects.theforeman.org/issues/8537 (please don't push commits to develop either right now).

**Comment by dLobatog** [comment]
[test]

**Comment by dLobatog** [comment]
Merged as 0f383a25b54af7e709b845f81074cfcbda644fd3, thanks @ohadlevy !

**Pull request closed without merging**